runtime/interpreter/value.go

```diff
@@ -1549,9 +1549,13 @@ func (v *ArrayValue) RecursiveString(seenReferences SeenReferences) string {
 	values := make([]string, v.Count())
 
 	i := 0
-	v.Walk(func(value Value) {
-		values[i] = value.RecursiveString(seenReferences)
+
+	_ = v.array.Iterate(func(element atree.Value) (resume bool, err error) {
+		// ok to not meter anything created as part of this iteration, since we will discard the result
+		// upon creating the string
```

Comment:
Is there a risk that Go's garbage collection won't catch this quickly enough?

My concern is that the memory is still being allocated even if it will (eventually) be deallocated so a well-crafted transaction could trigger this code path repeatedly. But that's just a guess since I don't know how this code is used now how Go actually does garbage collection.

Reply:
We still have to meter the amount of storage used before allocating though. The stored value could e.g. be a very large string value. 

`MustConvertUnmeteredStoredValue` should be `MustConvertStoredValue` with a memory gauge and pass it to `ConvertStoredValue`

Reply:
The issue with this is that this would require the `String` method on all values to take an `Interpreter` argument. Is this something we want?

Reply:
What do you all think of creating an issue for this and merging this PR in the current state? Since it will be an addition to the existing metering, we can get back to it once we have a conclusion.